### PR TITLE
feat(teams) Add team member count

### PR DESCRIPTION
**Pull request by markstory**
We'd like to show the number of members (active & inactive) in the team list page.

screen shot 2018-09-27 at 4 58 23 pm

Refs APP-75

### CODE REVIEW

File: src/sentry/api/serializers/models/team.py [hunk 1]
```diff
@@ -16,55 +17,70 @@
...
-            om.organization_id: om.role for om in
+        return {
+            om['organization_id']: om['role'] for om in
             OrganizationMember.objects.filter(
                 user=user,
                 organization__in=set(org_ids),
-            )}
-    else:
-        org_roles = {}
+            ).values('role', 'organization_id')
```
Comment (markstory) [inline src/sentry/api/serializers/models/team.py]: I switched this to a dict result as we were just tossing the ORM objects away.

File: tests/sentry/api/endpoints/test_organization_details.py [hunk 2]
```diff
@@ -48,7 +54,7 @@ def test_simple(self):
         )
         # TODO(dcramer): we need to pare this down -- lots of duplicate queries
         # for membership data
-        with self.assertNumQueries(28, using='default'):
+        with self.assertNumQueries(33, using='default'):
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_organization_details.py]: Not jazzed about this. I'd like to revisit this and see if I can shave a few queries off.

**Comment by billyvg** [comment]
What are the performance implications for orgs that have many teams?

**Comment by markstory** [comment]
@billyvg They will have an additional query per team in the response. I was not able to figure out how to get the results I wanted from django without an additional query.

**Comment by markstory** [comment]
@billyvg The additional query looks like

```sql
SELECT "sentry_team"."id", COUNT("sentry_organizationmember_teams"."id") AS "member_count" 
FROM "sentry_team" 
LEFT OUTER JOIN "sentry_organizationmember_teams" ON ( "sentry_team"."id" = "sentry_organizationmember_teams"."team_id" ) 
WHERE "sentry_team"."id" IN (<id list>) 
GROUP BY "sentry_team"."id", "sentry_team"."organization_id", "sentry_team"."slug", "sentry_team"."name", "sentry_team"."status", "sentry_team"."date_added"
```
and it will hit existing indexes.

```sql
HashAggregate  (cost=1843.76..1843.96 rows=20 width=51) |  
-- | --
->  Nested Loop Left Join  (cost=0.87..1840.81 rows=169 width=51) |  
->  Index Scan using sentry_team_pkey on sentry_team  (cost=0.44..60.83 rows=20 width=47) |  
Index Cond: (id = ANY ('{3,2,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24}'::bigint[])) |  
->  Index Scan  using sentry_organizationmember_teams_team_id on  sentry_organizationmember_teams  (cost=0.43..87.22 rows=178 width=8) |  
Index Cond: (sentry_team.id = team_id)
```

For ~20 teams the the estimated time is < 1ms.

**Comment by billyvg** [comment]
Maybe we can have @macqueen or @mattrobenolt weigh in on this.

UI wise the text looks odd floating in the middle of the row, what if we added it after the team name with our `user` icon? e.g. `(14 ICON)`

**Comment by markstory** [comment]
After talking with @billyvg latest UI looks like

screen shot 2018-10-03 at 2 37 41 pm

**Comment by Chrissy** [comment]
UI looks nice! It would look _even better_ if the size of the team avi matched the height of the two lines. This could be achieved by either bumping up the size of the avi or dropping the size of the text down.

**Comment by markstory** [comment]
@Chrissy I can do that. Does 

screen shot 2018-10-03 at 5 05 02 pm

look ok?

**Comment by EvanPurkhiser** [comment]
@markstory I think a few pixels larger on the avatar, It should look like the padding on the bottom of the member count matches the padding on top of the team name

**Review by evanpurkhiser — COMMENTED** [review]
Really love the refactoring of some of the queries to use `values` and `values_list`, great work. The query performance looks great too, thanks for the explain.

Approving for backend, but will leave to @Chrissy for the sizing of the avatar.

**Comment by Chrissy** [comment]
avatar looks good! if you wanted to be obsessive, it looks like it is maybe 2 pixels too high, but definitely not necessary.

**Comment by mattrobenolt** [comment]
I mean, the query itself is fine, but it'll be bad if teams have lots of members since COUNT() is just expensive to run or lots of teams since that's N COUNTs running. But it's pretty impossible to quantify at what point it'll be a problem, but high probability that this will break for someone. Can we pull and see teams with the most members in production and see what this query looks like for them? Maybe the upper end of "number of teams", and "teams with the most members." It's the only way we can really see how bad this will be. The query plan can't make this faster.

**Comment by markstory** [comment]
@mattrobenolt The customer with the largest number of teams (150 teams) generates the following query plan.

```
HashAggregate  (cost=6766.77..6768.27 rows=150 width=51) |  
-- | --
->  Nested Loop Left Join  (cost=1.03..6744.12 rows=1294 width=51) |  
->  Index Scan using sentry_team_pkey on sentry_team  (cost=0.60..412.21 rows=150 width=47) |  
Index Cond:  (id = ANY  ('{150 ids elided}'::bigint[])) |  
->  Index Scan  using sentry_organizationmember_teams_team_id on  sentry_organizationmember_teams  (cost=0.43..40.35 rows=186 width=8) |  
Index Cond: (sentry_team.id = team_id)

```

**Review by mattrobenolt — APPROVED** [review]
Approving on this while @markstory takes responsibility if it's slow. The query itself is fine and won't hurt infrastructure, it just adds additional overhead in a way that is linear complexity for the database. And in some edge cases, the query can add an additional 1 second or more to the response just to give you some numbers.

**Pull request merged**